pyccel/ast/numpyext.py

```diff
@@ -140,6 +140,11 @@ class NumpyReal(PythonReal):
     > np.real(a)
     1.0
     """
+    def __init__(self, arg):
+        super().__init__(arg)
+        self._precision = arg.precision
+        self._shape = process_shape(self.internal_var.shape)
+        self._rank  = len(self._shape)
```

Comment:
Does this not need doing for `NumpyImag` too?

Reply:
l thought , it will be done in another PR, what do you think ?

Reply:
There is very little difference between `NumpyReal` and `NumpyImag`. They are complimentary and it is very unexpected for one to implement more things than the other. This is why `PythonReal` and `PythonImag` have a common ancestor `PythonComplexProperty`, so I am not super comfortable with the idea of leaving this for later. Is it a lot of work?

Reply:
with tests ? yes, especially if we discovered something.

Reply:
> so I am not super comfortable with the idea of leaving this for later.
it will be done directly after this one to keep on track the theme of these two PRs.

Reply:
ok